docs/usage/self-hosted-experimental-flags.md

```diff
@@ -0,0 +1,130 @@
+# Self-hosted experimental flags
+
+The following flags are "experimental" because they:
+
+- are not commonly needed
+- are typically an effort to work around some other service's or platform's problem
+- can be removed at any time
+- are flags for Renovate's internal use to validate they work as intended
+
+Experimental flags which are commonly used and for which there is no external solution in sight can be converted to an official configuration option by the Renovate bot developers.
+
+Use these experimental flags at your own risk.
+These flags may be removed or have their behavior changed in **any** version.
+We will try to keep breakage to a minimum, but make no guarantees that an experimental flag will keep working.
+
+## `dockerHubTags`
+
+If added to the `experimentalFlags` list, Renovate will use the Docker Hub API (`https://hub.docker.com`) to fetch tags instead of the normal Docker API for images pulled from `https://index.docker.io`.
+
+Example usage:
+
+```js
+experimentalFlags: ['dockerHubTags'];
+```
+
+## `execGpidHandle`
+
+If added to the `experimentalFlags` list, Renovate will terminate the whole process group of a terminated child process spawned by Renovate.
+
+Example usage:
+
+```js
+experimentalFlags: ['execGpidHandle'];
+```
+
+## `noMavenPomCheck`
+
+If added to the `experimentalFlags` list, Renovate will skip its default artifacts filter check in the Maven datasource.
+Skipping the check will speed things up, but may result in versions being returned which don't properly exist on the server.
+
+Example usage:
+
+```js
+experimentalFlags: ['noMavenPomCheck'];
+```
+
+## `nugetDownloadNupkgs`
+
+If added to the `experimentalFlags` list, Renovate will download `nupkg` files for determining package metadata.
+
+Example usage:
+
+```js
+experimentalFlags: ['nugetDownloadNupkgs'];
+```
+
+## `paginateAll`
+
+If added to the `experimentalFlags` list, Renovate will always paginate requests to GitHub fully, instead of stopping after 10 pages.
+
+Example usage:
+
+```js
+experimentalFlags: ['paginateAll'];
+```
+
+## `rebasePaginationLinks`
+
+If added to the `experimentalFlags` list, Renovate will rewrite GitHub Enterprise Server's pagination responses to use the `endpoint` URL from the Renovate config.
+
+Example usage:
+
+```js
+experimentalFlags: ['rebasePaginationLinks'];
+```
+
+<!-- prettier-ignore -->
+!!! note
+    For the GitHub Enterprise Server platform only.
+
+## `repoCacheForceLocal`
+
+If added to the `experimentalFlags` list, Renovate will persist repository cache locally after uploading to S3.
+
+Example usage:
+
+```js
+experimentalFlags: ['repoCacheForceLocal'];
+```
+
+## `sqlitePackageCache`
+
+If added to the `experimentalFlags` list, Renovate will use SQLite as the backend for the package cache.
+Don't combine with `redisUrl`, Redis would be preferred over SQlite.
```

Comment:
```suggestion
Do not combine with `redisUrl`, as Renovate would then prefer Redis over SQlite.
```

Please check if this is correct. We should expand contractions like `don't` into `do not`. Also `Redis would be preferred over SQlite.` is vague about who prefers this.

I think we're trying to say something like:

> If you set `redisUrl` then Renovate ignores the `sqlitePackageCache` option, because it will use Redis instead of SQLite.